### ISSUE TITLE
Javamelody pointcuts only work if the target annotation is directly on the field or class

### ISSUE THREAD

**Issue by afillatre** [javamelody/javamelody#571]
I'm using Java Melody (JM) 1.60 with Spring Boot (SB) 1.2.3.

By default, JM monitors Spring Services (classes annotated with an `@Service` annotation. For some reasons, I had to create a child annotation, to annotate my classes. My custom annotation is annotated with the Spring `@Service` one. For Spring, my class is still a service, although it is not directly annotated with the `@Service` annotation (see the example below):

```java
@Service
public @interface MyCustomServiceAnnotation {
}

@ MyCustomServiceAnnotation
public class MyService {
   // For Spring, this class is a Service (i.e. a Spring bean)
}
```

It would be nice if JM could monitor my custom service, which is not the case ATM, because it isn't directly annotated with the Spring `@Service` annotation.
If I annotate my service with the `@MonitoredWithSpring` annotation, this works fine, but this is not a suitable solution for me.

Thanks for taking a look at this improvement !

**Comment by evernat** [comment]
This is a limitation of the Spring's AnnotationMatchingPointcut when using:
	@Bean
	public MonitoringSpringAdvisor springServiceMonitoringAdvisor() {
		final MonitoringSpringAdvisor interceptor = new MonitoringSpringAdvisor();
		interceptor.setPointcut(new AnnotationMatchingPointcut(Service.class));
		return interceptor;
	}
It does not match custom inherited annotations.

So I suggest that you add a new pointcut in JavaMelodyConfiguration with your custom annotation:
	@Bean
	public MonitoringSpringAdvisor springCustomServiceMonitoringAdvisor() {
		final MonitoringSpringAdvisor interceptor = new MonitoringSpringAdvisor();
		interceptor.setPointcut(new AnnotationMatchingPointcut(MyCustomServiceAnnotation.class));
		return interceptor;
	}

Otherwise I suppose that you can write you own [Pointcut ](http://grepcode.com/file/repo1.maven.org/maven2/org.springframework/spring/2.5.6/org/springframework/aop/Pointcut.java#Pointcut) implementation with your own [AnnotationClassFilter ](http://grepcode.com/file/repository.springsource.com/org.springframework/org.springframework.aop/3.2.0/org/springframework/aop/support/annotation/AnnotationClassFilter.java#AnnotationClassFilter.matches%28java.lang.Class%29) implementation.

**Comment by afillatre** [comment]
Thanks for your answer.

Unfortunately, both your suggestions won't work for me because of #570. This is actually what I already tried.

Taking another look at the `AnnotationMatchingPointcut` class, I see the constructor can take a 2nd parameter `checkInherited` that should do the trick. I tried it, but here again I think it doesn't work because of #570. It seems like although the configuration beans are loaded, they do not prevail against another configuration somehow loaded elsewhere.
